codec-http/src/main/java/io/netty/handler/codec/http/cookie/ServerCookieEncoder.java

```diff
@@ -141,19 +165,19 @@ public String encode(Cookie cookie) {
      * @param cookies a bunch of cookies
      * @return the corresponding bunch of Set-Cookie headers
      */
-    public List<String> encode(Collection<? extends Cookie> cookies) {
-        if (checkNotNull(cookies, "cookies").isEmpty()) {
+    public List<String> encode(Cookie... cookies) {
```

Comment:
I think it'll be easier in the long run to stick to the spec, cookie issues can be very hard to debug, and having everything that is built on Netty work in a nonstandard way will only contribute to the problem.

I just noticed though that I didn't read the spec closely enough (apologies), and I mistakenly applied the sorting to the ServerCookieEncoder. This sort order should actually be in the ClientCookieEncoder, since the spec is talking about the user agent, not the server. Do you want me to put together a PR for ClientCookieEncoder?

However, there is a related change that does need to be applied to ServerCookieEncoder: 

    Servers SHOULD NOT include more than one Set-Cookie header field in
    the same response with the same cookie-name.

This eliminates the need for sorting in the server, but there probably needs to be some sort of "last writer wins" heuristic here, whereby cookies are added to a HashMap keyed on cookie name as they are encoded, and then the values returned as a list after all cookies have been encoded.